pylint/checkers/refactoring.py

```diff
@@ -434,9 +439,19 @@ def _check_exception_inherit_from_stopiteration(exc):
         stopiteration_qname = '{}.StopIteration'.format(utils.EXCEPTIONS_MODULE)
         return any(_class.qname() == stopiteration_qname for _class in exc.mro())
 
-    @utils.check_messages('stop-iteration-return')
+    def _check_consider_using_dict_comprehension(self, node):
+        if (hasattr(node, 'func') and
```

Comment:
The node always has `func`, so the following should do. Note that its preferable to use `isinstance` against the correct type of node you expect rather than `hasattr`.
```
if isinstance(node.func, astroid.Name) and node.func.name == 'dict'
```